### PR TITLE
[U1] Frontend: Login - Validations

### PR DESCRIPTION
**Pull request by germangelv**
## Summary
- Added the basic validations to login and home view

## Trello Card
https://trello.com/c/d5MMUFtV/

## Screenshot
ezgif-2-6b9f0de1384f

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/RootActivity.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            activity = HomeActivity.class;
+        else
+            activity = LoginActivity.class;
+
+        intent = new Intent(this, activity);
+        this.intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        startActivity(intent);
+    }
+
+    private boolean OnUsernameLoginSaved() {
```
Comment (igsosa92) [inline app/src/main/java/ar/com/wolox/android/example/ui/RootActivity.java]: this should be called `isUsernameLoginSaved()`

File: app/src/main/java/ar/com/wolox/android/example/ui/RootActivity.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        return getUsernameLogin() != null;
+
+    }
+
+    private String getUsernameLogin() {
+        sharedPref = getSharedPreferences(getApplication().getString(R.string.vlogin_preferences_name), MODE_PRIVATE);
+        return sharedPref.getString("username", null);
```
Comment (igsosa92) [inline app/src/main/java/ar/com/wolox/android/example/ui/RootActivity.java]: `"username"` should be define as a constant value

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 3]
```diff
@@ -33,25 +34,43 @@ public void init() {
         ButterKnife.bind(this, getActivity());
     }
 
+
```
Comment (igsosa92) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: delete this blank line

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 4]
```diff
@@ -33,25 +34,43 @@ public void init() {
...
-                else if (vLogInPassword.getText().toString().isEmpty())
-                    Toast.makeText(requireContext(), "Ingrese Contraseña valida", Toast.LENGTH_SHORT).show();
-                else if (vLogInPassword.getText().toString().length() < 6)
-                    Toast.makeText(requireContext(), "La contraseña debe tener mas de 6 caracteres", Toast.LENGTH_SHORT).show();
-                else
-                    Toast.makeText(requireContext(), "Login presionado", Toast.LENGTH_SHORT).show();
+                if (areValidFields())
+                    getPresenter().doLogin(vLogInEmail.getText().toString(), vLogInPassword.getText().toString());
             }
+
```
Comment (igsosa92) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: delete this blank line

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 5]
```diff
@@ -33,25 +34,43 @@ public void init() {
...
+    @Override
+    public void onLoginSuccess() {
+        getPresenter().setPreferences(getContext(), vLogInEmail.getText().toString());
+        Intent intent = new Intent(getActivity(), HomeActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        startActivity(intent);
+    }
+
+
+    private boolean areValidFields() {
```
Comment (igsosa92) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: this should be done in the presenter

File: app/src/main/res/layout/fragment_login.xml [hunk 6]
```diff
@@ -36,7 +36,7 @@
         android:layout_marginTop="16dp"
         android:layout_marginRight="16dp"
         android:gravity="center"
-        android:hint="@string/login_email"
+        android:hint="@string/vlogin_email"
```
Comment (igsosa92) [inline app/src/main/res/layout/fragment_login.xml]: rename this to `login_email`

File: app/src/main/res/layout/fragment_home.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:orientation="vertical">
+
+    <TextView
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:gravity="center"
+        android:text="@string/vhome__activity"
```
Comment (igsosa92) [inline app/src/main/res/layout/fragment_home.xml]: rename this to `home_activity`

File: app/src/main/res/layout/fragment_login.xml [hunk 8]
```diff
@@ -61,7 +61,7 @@
         android:layout_marginRight="16dp"
         android:background="@color/wolox_white"
         android:gravity="center"
-        android:text="@string/login_button_login" />
+        android:text="@string/vlogin_button_login" />
```
Comment (igsosa92) [inline app/src/main/res/layout/fragment_login.xml]: remember to rename everything without the 'v' at the beginning

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java (hunk 5 above)
Comment (germangelv) [reply app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: I rename now the method areValidFields() to ValidationFields in LoginPresenter

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java [hunk 9]
```diff
@@ -30,21 +29,17 @@ public int layout() {
...
 
-
     @Override
     public void setListeners() {
-
         vLogInButton.setOnClickListener(new View.OnClickListener() {
             @OnClick
             public void onClick(View v) {
-                if (areValidFields())
+                if (getPresenter().ValidationFields(vLogInEmail.getText().toString(), vLogInPassword.getText().toString()))
```
Comment (igsosa92) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginFragment.java]: It is bad, but not that bad xD I would do something like: getPresenter.ValidationFields(fafafa). Then in the presenter: validationFields(){ if ( itIsEverythingOK) doLogin(), else getView().showError.
Also remember that functions names are in camelCase and not in PascalCase. Always begins in small letter.

File: app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java [hunk 10]
```diff
@@ -18,17 +19,26 @@
...
 
     public void doLogin(String emailField, String passwordField) {
         //Logica para hacer el login al Back-END
         getView().onLoginSuccess();
-
     }
 
+    public boolean ValidationFields(String emailField, String passwordField) {
+        if (emailField.isEmpty())
+            getView().onEmailFieldEmpty();
```
Comment (igsosa92) [inline app/src/main/java/ar/com/wolox/android/example/ui/login/LoginPresenter.java]: This is OK and it's just a commet, but do you think is there a way to implement this to call getView().showValidationError(case) ? the you don't have to implement a method for each error case. Again, It's just a comment

File: app/src/main/res/values/strings.xml [hunk 11]
```diff
@@ -12,20 +12,19 @@
...
+    <string name="login_empty_password">El campo de contraseña es requerido</string>
+    <string name="login_invalid_email">Correo electronico invalido</string>
+    <string name="login_email">example@domain.com</string>
+    <string name="login_preferences_name">ar.com.wolox.android.example.ui.login</string>
+    <string name="login_username">usuario</string>
+    <string name="login_password">contraseña</string>
+    <string name="login_button_login">LOG IN</string>
+    <string name="login_button_signup">SIGN UP</string>
+    <string name="login_terms_and_conditions">By joing you agree to our <u>Terms &amp; Conditions</u></string>
+    <string name="home__activity">HOME</string>
```
Comment (igsosa92) [inline app/src/main/res/values/strings.xml]: delete one underscore, it's just `home_activity`

File: app/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -12,20 +12,19 @@
...
     <string name="password_required">Password is Required</string>
-
+    <string name="vRootUsername">username</string>
```
Comment (igsosa92) [inline app/src/main/res/values/strings.xml]: This should not be saved as a string resource. It should be a constant value inside the class that implement it.

**Pull request merged**